quipucords/api/deployments_report/tasks.py

```diff
@@ -3,17 +3,49 @@
 import logging
 
 import celery
-
-from api.deployments_report.model import DeploymentsReport
-from api.deployments_report.util import create_deployments_csv
-from api.deployments_report.view import build_cached_json_report
+from django.db import transaction
 
 logger = logging.getLogger(__name__)
 
 
 @celery.shared_task()
+@transaction.atomic
+def generate_cached_fingerprints(deployments_report_id: int):
+    """Generate cached fingerprints based on saved SystemFingerprint objects."""
+    from api.deployments_report.model import DeploymentsReport, SystemFingerprint
+
+    try:
+        from api.deployments_report.serializer import SystemFingerprintSerializer
+
+        deployments_report = DeploymentsReport.objects.get(id=deployments_report_id)
+        deployments_report.cached_csv_file_path = None  # cached CSV may now be invalid
+        deployments_report.cached_fingerprints = SystemFingerprintSerializer(
+            SystemFingerprint.objects.filter(deployment_report=deployments_report),
+            many=True,
+        ).data
+        deployments_report.save()
```

Comment:
Huh, my memory played me here. I thought we were using a json file stored on the filesystem.

Reply:
From current version in `main`:

https://github.com/quipucords/quipucords/blob/main/quipucords/fingerprinter/runner.py#L248
https://github.com/quipucords/quipucords/blob/main/quipucords/fingerprinter/runner.py#L290

Note that we've historically just assigned a `list` of `dict`s directly to `cached_fingerprints`, and we have not converted or sanitized it in some way for future JSON output. I think we've just been lucky that it's already JSON-safe with no troublesome `datetime` or other objects. 🤷

Reply:
We do write it as JSON when it finally goes to the filesystem, but not anywhere before then:

https://github.com/quipucords/quipucords/blob/main/quipucords/api/deployments_report/model.py#L99-L105